src/Component/RuleGenerator/RuleGenerator.tsx

```diff
@@ -0,0 +1,398 @@
+    this.setState({classificationMethod});
+  }
+
+  onNumberChange = (numberOfRules: number) => {
+    let {
+      colorRamp,
+      colorRampName
+    } = this.state;
+
+    if (numberOfRules <= 2) {
+      colorRamp = undefined;
+      colorRampName = undefined;
+    } else if (colorRampName) {
+      colorRamp = RuleGeneratorUtil.getColorRamp(colorRampName, numberOfRules);
+      if (!colorRamp) {
+        colorRampName = undefined;
+      }
+    }
+
+    this.setState({
+      numberOfRules,
+      colorRamp,
+      colorRampName
+    });
+  }
+
+  getBackgroundStyleFromColors = (colors: string[]) => {
+    const gradients = colors.map((color: string) => `linear-gradient(${color}, ${color})`);
+    const backgroundImage = gradients.join(',');
+    const size = colors.map((color: string, index: number) => {
+      const width = (index + 1) * (100 / colors.length);
+      return `${width}% 100%`;
+    });
+    const backgroundSize = size.join(',');
+    return {
+      backgroundImage: backgroundImage,
+      backgroundSize: backgroundSize,
+      backgroundRepeat: 'no-repeat'
+    };
+  }
+
+  getClassificationOptions = () => {
+    const {
+      classificationMethods,
+      locale
+    } = this.props;
+    return classificationMethods.map((classificationMethod: ClassificationMethod) => {
+      return (
+        <Select.Option
+          className="classification-option"
+          key={classificationMethod}
+          value={classificationMethod}
+        >
+          {locale[classificationMethod] || classificationMethod}
+        </Select.Option>
+      );
+    });
+  }
+
+  getColorOptions = () => {
+    const {
+      numberOfRules
+    } = this.state;
+
+    const {
+      colorRampNames
+    } = this.props;
+
+    const filteredColorMaps = colorRampNames.filter((colorRampName: string) => {
+      const minClasses = _get(colorScales, `[${colorRampName}].length`);
+      return numberOfRules > minClasses;
+    });
+
+    return filteredColorMaps.map((colorRampName) => {
+      const colors = RuleGeneratorUtil.getColorRamp(colorRampName, numberOfRules);
+      const style = this.getBackgroundStyleFromColors(colors);
+      return (
+        <Select.Option
+          className="color-ramp-option"
+          key={colorRampName}
+          value={colorRampName}
+          style={style}
+        >
+          {colorRampName}
+        </Select.Option>
+      );
+    });
+  }
+
+  onColorRampNameChange = (colorRampName: string) => {
+    const {
+      numberOfRules
+    } = this.state;
+    const colorRamp = RuleGeneratorUtil.getColorRamp(colorRampName, numberOfRules);
+    this.setState({
+      colorRampName,
+      colorRamp
+    });
+  }
+
+  onSymbolizerKindChange = (symbolizerKind: SymbolizerKind) => {
+    const {
+      wellKnownName
+    } = this.state;
+    this.setState({
+      symbolizerKind,
+      wellKnownName: symbolizerKind === 'Mark' ? wellKnownName : undefined
```

Comment:
If I get it right, once `symbolizerKind` was set to something unequal to `'Mark'`, `wellKnownName` will always be `undefined`.